File: cpp/base/src/main/vsdk/toolkit/common/statistics/SolidTextureStatistics.cpp
#include "java/util/ArrayList.txx"
#include "vsdk/toolkit/common/statistics/SolidTextureStatistics.h"

SolidTextureStatistics::SolidTextureStatistics(
    java::ArrayList<SolidTextureStatistics *> *partsPerThread)
{
    reset();

    if (partsPerThread == nullptr) {
        return;
    }

    for (long i = 0; i < partsPerThread->size(); ++i) {
        const SolidTextureStatistics *part = partsPerThread->get(i);
        if (part == nullptr) {
            continue;
        }
        callsToNoise += part->callsToNoise;
        callsToDNoise += part->callsToDNoise;
    }
}

File: java/base/src/main/vsdk/toolkit/common/statistics/SolidTextureStatistics.java
package vsdk.toolkit.common.statistics;

import java.util.List;

public final class SolidTextureStatistics
{
    public long callsToNoise;
    public long callsToDNoise;

    public SolidTextureStatistics()
    {
        reset();
    }

    public SolidTextureStatistics(List<SolidTextureStatistics> partsPerThread)
    {
        reset();
        if ( partsPerThread == null ) {
            return;
        }
        for ( SolidTextureStatistics part : partsPerThread ) {
            if ( part != null ) {
                callsToNoise += part.callsToNoise;
                callsToDNoise += part.callsToDNoise;
            }
        }
    }

    public void reset()
    {
        callsToNoise = 0L;
        callsToDNoise = 0L;
    }
}
